js/style-spec/function/index.js

```diff
@@ -86,34 +108,30 @@ function createFunction(parameters, defaultType) {
             }
 
             for (const z in featureFunctions) {
-                featureFunctionStops.push([featureFunctions[z].zoom, createFunction(featureFunctions[z])]);
+                featureFunctionStops.push([featureFunctions[z].zoom, createFunction(featureFunctions[z], propertySpec)]);
             }
             fun = function(zoom, feature) {
                 return outputFunction(evaluateExponentialFunction({
                     stops: featureFunctionStops,
                     base: parameters.base
-                }, zoom)(zoom, feature));
+                }, propertySpec, zoom)(zoom, feature));
```

Comment:
Why do you need to pass `propertySpec` to `evaluateExponentialFunction` both as an argument and embedded within `featureFunctionStops`?